code/game/objects/items/weapons/twohanded.dm

```diff
@@ -209,6 +209,42 @@
 /obj/item/twohanded/fireaxe/boneaxe/update_icon()
 	icon_state = "bone_axe[wielded]"
 
+/obj/item/twohanded/fireaxe/energized
+	desc = "Someone with a love for fire axes decided to turn this one into a high-powered energy weapon. Seems excessive."
+	force_wielded = 30
+	armour_penetration = 20
+	var/charge = 30
+	var/max_charge = 30
+
+/obj/item/twohanded/fireaxe/energized/update_icon()
+	if(wielded)
+		icon_state = "fireaxe2"
+	else
+		icon_state = "fireaxe0"
+
+/obj/item/twohanded/fireaxe/energized/New()
+	..()
+	START_PROCESSING(SSobj, src)
+
+/obj/item/twohanded/fireaxe/energized/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	return ..()
+
+/obj/item/twohanded/fireaxe/energized/process()
+	charge = min(charge + 1, max_charge)
```

Comment:
Would be better to just use a time instead of it to process constantly.
Currently this takes 30 * 20 (wait time of SSobj) ticks to gain full charge.

You can also make use of a callback if you want to give the user feedback as to when it is charged again

Reply:
Don't do either; just set it so that you can only use it once every 60 seconds by setting a cooldown of worldtime + 600---then have it display a message if you try to use it before then.

Reply:
Using it is attacking something right? So the user gets spammed 60 seconds long when he attacks something that the axe is not charged yet. Not sure if that is a good addition since the user can't check prior to attacking something. And attacking is quite a thing